src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java

```diff
@@ -91,6 +91,8 @@
 
     private String resourceLimitMemory;
 
+    private String restartPolicy = "Never";
```

Comment:
You need to move to `readResolve` the initialization, because on upgrade `restartPolicy` instance variable will be null